dhall-docs/src/Dhall/Docs.hs

```diff
@@ -162,32 +178,57 @@ saveHtml inputAbsDir outputAbsDir packageName t@(absFile, _) = do
 
     let relativeResourcesPath = resolveRelativePath outputAbsDir htmlOutputDir
 
+    let strippedHeader = stripCommentSyntax header
+    headerAsHtml <- case markdownToHtml absFile strippedHeader of
+        Left err -> do
+            putStrLn $ markdownParseErrorAsWarning err
+            return $ Lucid.toHtml strippedHeader
+        Right html -> return html
+
     Lucid.renderToFile (Path.fromAbsFile htmlOutputFile)
-        $ filePathHeaderToHtml t DocParams {..}
+        $ filePathHeaderToHtml (absFile, headerAsHtml) DocParams {..}
 
     return htmlOutputFile
   where
     addHtmlExt :: Path b File -> IO (Path b File)
     addHtmlExt = Path.addExtension ".html"
 
+    markdownParseErrorAsWarning :: MarkdownParseError -> String
+    markdownParseErrorAsWarning MarkdownParseError{..} =
+        "\n\ESC[1;33mWarning\ESC[0m\n\n" <>
+        Text.Megaparsec.errorBundlePretty unwrap <>
+        "The original non-markdown text will be pasted in the documentation"
+
+    stripCommentSyntax :: Header -> Text
+    stripCommentSyntax (Header h)
+        | Just s <- Data.Text.stripPrefix "--" strippedHeader
+            = s
+        | Just commentPrefixStripped <- Data.Text.stripPrefix "{-" strippedHeader
+        , Just commentSuffixStripped <- Data.Text.stripSuffix "-}" commentPrefixStripped
+            = commentSuffixStripped
+        | otherwise = strippedHeader
```

Comment:
In which scenario are we hitting this branch?

Reply:
I think that in this moment no scenario is going there - I tend to always fill all the possible branches. I can remove it if you think its better.

Reply:
If the branch would only be used in case of a programmer mistake, we could add something like this:

```suggestion
        | otherwise = error $ "Unexpected header:\n" ++ Text.unpack h
```